kazoo/protocol/connection.py

```diff
@@ -247,7 +248,14 @@ def _read(self, length, timeout):
                         # have to check wlist and xlist as we don't set any
                         raise self.handler.timeout_exception(
                             "socket time-out during read")
-                chunk = self._socket.recv(remaining)
+                try:
+                    chunk = self._socket.recv(remaining)
+                except ssl.SSLError as e:
+                    if e.errno in (ssl.SSL_ERROR_WANT_READ,
+                                   ssl.SSL_ERROR_WANT_WRITE):
```

Comment:
Does it make sense to check for `WANT_WRITE` in the middle of a `recv()` call? Should the SSL try / catch not be aligned to the operation we are undertaking?

Reply:
Yes, sorry I linked to the docs in the Issue but didn't here in the PR; this link should explain that due to the way OpenSSL's state machine works, either result can happen on either kind of underlying call: https://www.openssl.org/docs/manmaster/man3/SSL_get_error.html

Reply:
Ok, thank you for clarifying. Makes sense